api/src/main/java/org/openmrs/hl7/handler/ORUR01Handler.java

```diff
@@ -359,9 +359,11 @@ private Message processORU_R01(ORU_R01 oru) throws HL7Exception {
 				finally {
 					// Handle obs-level exceptions
 					if (errorInHL7Queue != null) {
-						throw new HL7Exception(Context.getMessageSourceService().getMessage(
-						    "ORUR01.error.improperlyFormattedOBX",
-						    new Object[] { PipeParser.encode(obx, new EncodingCharacters('|', "^~\\&")) }, null),
+						throwHL7Exception(String message, int errorCondition, Throwable cause) {
+		                   super(message, cause);
+		                  setErrorCode(errorCondition);   
+						
+							new Object[] { PipeParser.encode(obx, new EncodingCharacters('|', "^~\\&")) }, null),
 						        HL7Exception.DATA_TYPE_ERROR, errorInHL7Queue);
```

Comment:
Hii @Devanshk9, the travis-ci build is still failing.

To solve this issue, you need to return an `HL7Exception` using the new constructor with takes `ErrorCode error` object as second parameter instead of simple `int errorCode`  as I mentioned in previous review